src/classes/CountertopWorker.js

```diff
@@ -0,0 +1,132 @@
+import { IAppliance } from '@tvkitchen/base-interfaces'
+import { Payload } from '@tvkitchen/base-classes'
+import { applianceEvents } from '@tvkitchen/base-constants'
+import { consoleLogger } from '../tools/loggers'
+import kafka from '../tools/kafka'
+import { getStreamTopic } from '../tools/utils/countertop'
+
+/**
+ * A CountertopWorker (aka Line Cook) is responsible for monitoring specific data streams
+ * and processing those streams using TV Kitchen Appliances to create new data.
+ *
+ * CountertopWorker instances consume Payloads from Kafka and pass them to their Appliances.
+ * CountertopWorker instances listen for emitted Payloads from appliances and process them.
+ */
+class CountertopWorker {
+	logger = null
+
+	consumer = null
+
+	producer = null
+
+	admin = null
+
+	appliance = null
+
+	stream = null
+
+	/**
+	 * Create a new CountertopWorker.
+	 *
+	 * @param  {Class}            Appliance     The class for the appliance that this worker will use.
+	 * @param  {CountertopStream} stream        A stream that ends in this worker's station.
+	 * @param  {Object}           configuration Configuration settings for the Appliance.
+	 */
+	constructor(
+		Appliance,
+		applianceSettings = {},
+		{
+			stream,
+			logger = consoleLogger,
+		} = {},
+	) {
+		this.stream = stream
+		this.logger = logger
+		this.consumer = kafka.consumer({ groupId: stream.id })
+		this.producer = kafka.producer()
+		this.admin = kafka.admin()
+		this.appliance = new Appliance(applianceSettings)
+		if (!IAppliance.isIAppliance(this.appliance)) {
+			throw new Error('TVKitchen can only use Appliances that extend IAppliance.')
+		}
+
+		this.appliance.on(
+			applianceEvents.PAYLOAD,
+			(payload) => this.producer.send({
+				topic: getStreamTopic(payload.type, this.stream),
+				messages: [{
+					value: Payload.serialize(payload),
+				}],
+			}),
+		)
+	}
+
+	/**
+	 * Start the worker and begin data processing.
+	 *
+	 * @return {Boolean} Whether the worker successfully started.
+	 */
+	start = async () => {
+		this.logger.trace(`CountertopWorker<${this.stream.id}>: start()`)
+		if (!await this.appliance.audit()) {
+			return false
+		}
+		if (!this.appliance.start()) {
+			return false
+		}
+		await this.admin.connect()
+		await this.producer.connect()
+		await this.consumer.connect()
+		const inputTopics = this.appliance.constructor.getInputTypes().map(
```

Comment:
Maybe should be a named utility

Reply:
Deciding to not bike shed this class yet since I expect it to be in a bit of flux.  I think both `start` and `stop` are a doing a bit too much without relying on utility methods.